### ISSUE TITLE
Read image Error failed

### ISSUE THREAD

**Issue by charlottte** [towhee-io/examples#224]
2023-08-03 14:01:57,823 - 139852063110912 - image_decode_cv2.py-image_decode_cv2:40 - ERROR: Download image from http://*******904d20239a385ec9.jpg failed, error msg: {"Code":"NoSuchURL","Msg":"The specified URL does not exist."}, request code: 404
2023-08-03 14:01:57,823 - 139852063110912 - image_decode_cv2.py-image_decode_cv2:68 - ERROR: Read image http://*******/904d20239a385ec9.jpg failed
when i use pipe to get image feature,if the url link failure，The pipe program will jam without export nothing
![image](https://github.com/towhee-io/examples/assets/29828790/7439cba2-b1b1-4776-b881-8282d084cbde)

**Comment by junjiejiangjjj** [comment]
You can use  `pstack` or `strace` to see where the program is hanging

**Comment by charlottte** [comment]
i find the function drop_emtpy in the Early version,but i don't know how to use it in the latest version

**Comment by charlottte** [comment]
i find the function drop_emtpy in the Early version,but i don't know how to use it in the latest version,how to use it with pipe

**Comment by junjiejiangjjj** [comment]
If an error occurs during execution, it will raise an exception. If you want to ignore the exception, you just need to catch it
```python
try:
     # run pipeline
except:
    pass

**Comment by charlottte** [comment]
Sorry, let me rephrase my question. I currently have a CSV file with URLs of images and their corresponding text descriptions. I would to extract features from both the images and text, then merge, and finally store them in a database. However, I am facing a few issues. Firstly, when reading the URLs, some of them may be failure. How can I implement handling to continue the program execution? Secondly, how can I perform batch processing? Are there any relevant examples available?

**Comment by junjiejiangjjj** [comment]
The pipeline running error will only throw exceptions but won't hang, so we need to check the cause of the hang

**Comment by junjiejiangjjj** [comment]
https://towhee.io/data-source/csv-reader  Here are some examples related to processing CSV and Batch

**Comment by charlottte** [comment]
thanks a lot!I've solved the hanging problem。BUT there is another questions，when i use batch, the Processing is not speeding up，can batch use with gpu ，is there any example

**Comment by junjiejiangjjj** [comment]
You can use the debug interface to see where the performance bottleneck is， https://towhee.readthedocs.io/en/latest/debug/usage.html#usage

**Comment by junjiejiangjjj** [comment]
The slowness could be caused by the image downloading and decoding.
